### PR TITLE
Issue #1675 report failed time entries

### PR DESCRIPTION
**Pull request by timstallmann**
@chrisarusso assigning to you for a once-over, feel free to pass along if you think someone else is better-equipped. Once this is merged the docker hub image will automatically rebuild and we'll have to touch base about updating the cron job you have to run docker instead of PHP.

To test:
* Spin up [Redmine](https://github.com/savaslabs/redmine_savas) locally using Docker and the instructions in the readme
* copy config.example.yml to config.yml
* In Config.yml:
* Set `debug-user` under `slack` to your slack name (e.g. `@tim`)
* Set the redmine URL to `http://local.pm.savaslabs.com`
* Log into your local redmine instance (same username/password as production redmine), click on "My Account", click "show" under "API Access Key" and copy that API Access Key into the Redmine API Access Key in config.yml
* Get the slack webhook URL from @timstallmann and fill that into config.yml
* Run `docker build -t savaslabs/sumac:test .` in the repo directory to build the docker image
* Use the run command examples in the readme to run sync, but add the `:test` tag to the docker image name to make sure you're using your local image.

### CODE REVIEW

**Comment by lrowang** [comment]
Hi @timstallmann -

Unfortunately, the script isn't fully working for me. **It works fine if there are no "bad" time entries, but it will fail if I introduce a "bad" time entry with a made-up issue# and spelling mistakes.** I tried this for multiple different dates and this is what happened:

- The script will time out and give me a "sync" error and I won't get a slackbot notification. See **Pic1**.
- The script is only catching the "wrong issue#" error. If I remove that and just leaving the spelling mistakes then it runs smoothly (but doesn't catch the error). See **Pic2**.

When it comes to this PR review, I think I'm only going to be useful for testing. It'll probably be good if someone on the team with a better understanding can do the actual "code review". That said, let me know if you want me to continue testing different things! 

### Pic1
sumac-issue


### Pic2
sumac-spell

**Comment by timstallmann** [comment]
@lrowang what sorts of slack notifications did you get? What you're describing in terms of the command line output sounds like what the scripts behavior should be, although it raises a good question about what should cause command line errors (e.g. currently spelling errors don't, they just go to slack)

@chrisarusso could you weigh in on when you'd like to get command-line notifications for malformed time entries from the Sumac script vs. when they should just go to Slack notifications? Currently, the following just go to Slack and no Sumac error is logged/output
* Spelling mistakes
* Missing issue number

The following go to Slack and an error is logged/output by Sumac
* Issue does not exist in Redmine
* Issue exists but is in the wrong project

**Comment by lrowang** [comment]
@timstallmann - I didn't get any slackbot notifications. I was expecting on for the scenario in Pic1, but it didn't arrive. I tried a couple different dates and had the same result 😞 .

File: README.md [hunk 1]
```diff
@@ -4,39 +4,39 @@
...
-If you are developing, you'll want to run `composer install` on the host, then:
+Spin up redmine locally.
 
-```
-docker run -e SUMAC_HARVEST_MAIL=kosta@savaslabs.com --net redmine_default -it --rm -v $(pwd):/usr/src/sumac savaslabs/sumac sync -u 20160915:20160915
-```
+Copy `config.example.yml` to `config.yml` and fill in any values (particularly, you'll need the Slack webhook URL if you want to test slack integration).
 
-You can also use environment variables to set the configuration:
+Within the `sumac` directory, run `docker run --net redmine_default -it --rm -v $(pwd):/usr/src/sumac savaslabs/sumac sync -c /tmp/sumac/config.yml -u 20160915:20160916`.
```
Comment (dmurphy1) [inline README.md]: Based on the Dockerfile which has `COPY . /usr/src/sumac`, shouldn't `/tmp/sumac/config.yml` be `/usr/src/sumac/config.yml`?

**Comment by timstallmann** [comment]
@dmurphy1 ohh good catch! Actually the mistake here is that the `-v` 
flag to docker run should be `-v $(pwd):/tmp/sumac/`, not 
`/usr/src/sumac`. And then the `-c` flag to `sync` loads the custom 
config file from the volume mounted at `/tmp/sumac`. That way if you're 
just running the docker image you don't end up overwriting all the code 
which is already in `/usr/src/sumac` inside the container.

Dan Murphy wrote:

**Comment by dmurphy1** [comment]
After testing and reviewing the code, this looks good to me! Really nice work @timstallmann - this is an awesome feature! 👍 🚀 🎆

**Pull request merged**